path/_os.ts

```diff
@@ -1,35 +1,78 @@
 // Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
 // This module is browser compatible.
 
-// Keep this up-to-date with Deno.build.os
+export const isWindows: boolean = getIsWindows() === true;
+
+function getIsWindows(): boolean | undefined {
+  return (
+    getIsWindowsOnDeno() || getIsWindowsOnBrowser() || getIsWindowsOnNodeOrBun()
+  );
+}
+
 /**
- * Operating system type, equivalent to the type of
- * {@linkcode https://deno.land/api?s=Deno.build | Deno.build.os}.
+ * @returns whether the os is windows or undefined if not running
+ * in a deno runtime, undefined if not a deno runtime.
  */
-type OSType =
-  | "darwin"
-  | "linux"
-  | "windows"
-  | "freebsd"
-  | "netbsd"
-  | "aix"
-  | "solaris"
-  | "illumos";
-
-const osType: OSType = (() => {
+function getIsWindowsOnDeno(): boolean | undefined {
   // deno-lint-ignore no-explicit-any
   const { Deno } = globalThis as any;
   if (typeof Deno?.build?.os === "string") {
-    return Deno.build.os;
+    return Deno.build.os === "windows";
   }
+}
 
+/**
+ * @returns whether the os is windows or undefined if not running
+ * in a web browser, undefined if not a web browser
+ */
+function getIsWindowsOnBrowser(): boolean | undefined {
   // deno-lint-ignore no-explicit-any
   const { navigator } = globalThis as any;
-  if (navigator?.appVersion?.includes?.("Win")) {
-    return "windows";
+
+  return containsWindows(navigator?.userAgent);
+}
+
+/**
+ * according to documentation node's os module is implemented
+ * in bun as well.
+ * {@link https://bun.sh/docs/runtime/nodejs-apis#node-os}
+ *
+ * @returns whether the os is windows or undefined if not running
+ * on node or bun runtime
+ */
+function getIsWindowsOnNodeOrBun() {
+  const os = tryGettingNodeOsModule();
+
+  return containsWindows(os?.version());
+}
+
+type OsModule = {
+  version(): string;
+};
+
+// deno-lint-ignore no-explicit-any
+declare const require: any;
+
+function tryGettingNodeOsModule(): OsModule | undefined {
+  try {
+    return getNodeOsModule();
+  } catch {
+    return undefined;
   }
+}
 
-  return "linux";
-})();
+function getNodeOsModule(): OsModule | undefined {
+  if (require !== undefined) {
+    return require("os");
```

Comment:
If node.js is in ESM mode (.mjs or `type: module` in package.json), then it doesn't have `require`.

Reply:
Ok, I tested by manually converting to js and then running with `node os.js`, explains why this worked for me then.

I'll mark this PR as Draft again until I can come up with a better solution.